community/spatial-index/src/test/java/org/neo4j/gis/spatial/index/curves/SpaceFillingCurveTest.java

```diff
@@ -338,39 +356,209 @@ public void shouldGet2DHilbertSearchTilesForManyLevels()
+                                {
+                                    HistogramMonitor monitor = new HistogramMonitor( curve.getMaxLevel() );
+                                    final double xStart = xmin + xOffset;
+                                    final double xEnd = xStart + xExtent;
+                                    final double yStart = ymin + yOffset;
+                                    final double yEnd = yStart + yExtent;
+                                    Envelope searchEnvelope = new Envelope( xStart, xEnd, yStart, yEnd );
+                                    final long start = System.currentTimeMillis();
+                                    List<SpaceFillingCurve.LongRange> ranges = curve.getTilesIntersectingEnvelope( searchEnvelope, config, monitor );
+                                    final long end = System.currentTimeMillis();
+                                    debug( String.format(
+                                            "Results for level %d, with x=[%f,%f] y=[%f,%f]. Search size vs covered size: %d vs %d (%f x). Ranges: %d. Took %d ms\n",
+                                            level, xStart, xEnd, yStart, yEnd, monitor.getSearchArea(), monitor.getCoveredArea(),
+                                            (double) (monitor.getCoveredArea()) / monitor.getSearchArea(), ranges.size(), end - start ) );
+                                    int[] counts = monitor.getCounts();
+                                    for ( int i = 0; i <= monitor.getHighestDepth(); i++ )
+                                    {
+                                        debug( "\t" + i + "\t" + counts[i] );
+                                    }
+
+                                    areaStats.add( (double) (monitor.getCoveredArea()) / monitor.getSearchArea() );
+                                    rangeStats.add( ranges.size() );
+                                    maxDepthStats.add( monitor.getHighestDepth() );
+
+                                    assertThat( String.format( "Search size was bigger than covered size for level %d, with x=[%s,%s] y=[%s,%s]", level,
+                                            Double.toHexString( xStart ), Double.toHexString( xEnd ), Double.toHexString( yStart ),
+                                            Double.toHexString( yEnd ) ), monitor.getSearchArea(), lessThanOrEqualTo( monitor.getCoveredArea() ) );
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // Average over all runs on this level
+                log( String.format( formatBody, level, config.toString(),
+                        areaStats.avg(), areaStats.min, areaStats.max,
+                        rangeStats.avg(), rangeStats.min, rangeStats.max,
+                        maxDepthStats.avg(), maxDepthStats.min, maxDepthStats.max ) );
+            }
+        }
+    }
+
+    @Ignore
+    public void debugSingle()
```

Comment:
Why is this test ignored? A TODO comment about when it should be un-ignored would be good